src/engine/strat_engine/blockdevmgr.rs

```diff
@@ -40,17 +42,50 @@ pub fn resolve_devices(paths: &[&Path]) -> io::Result<HashSet<Device>> {
         .collect()
 }
 
+#[derive(Debug)]
+pub struct BlkDevSegment {
```

Comment:
It's not obvious to me that:

```
pub struct BlkDevSegment {
  pub uuid: Uuid,
  pub device: Device,
  pub start: Sectors,
  pub length: Sectors,
}
```

wouldn't be just as useful and would also eliminate ref counting requirement.